Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -25,12 +25,105 @@
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
 
+#include "vtkExtentTranslator.h"
+
+
+
+//---- option 2
+
+     #include <tbb/parallel_for.h>
+#include <tbb/blocked_range.h>
+#include <tbb/blocked_range3d.h>
+#include <tbb/blocked_range2d.h>
+#include <tbb/task_scheduler_init.h>
+//-- option 1
+#include "vtkSMPTools.h"
+
+
+
+
+
+
+struct vtkImageThreadStruct
+{
+vtkThreadedImageAlgorithm *Filter;
+vtkInformation *Request;
+vtkInformationVector **InputsInfo;
+vtkInformationVector *OutputsInfo;
+vtkImageData   ***Inputs;
+vtkImageData   **Outputs;
+};
+
+class apply_transform{
+  
+public:
+  vtkImageThreadStruct* str;
+  apply_transform(vtkImageThreadStruct* abc)
+  {
+    str = abc;
+  }
+  apply_transform()
+  {
+   
+  }
+  void operator()(const tbb::blocked_range2d<vtkIdType>& r) const {
+
+    int extent[6];
+    extent[0] = r.cols().begin();
+    extent[1] = r.cols().end();
+    extent[2] = r.rows().begin();
+    extent[3] = r.rows().end();
+    extent[4] = 0;
+    extent[5] = 0;    
+    //r.cols
+    str->Filter->ThreadedRequestData(str->Request,
+      str->InputsInfo, str->OutputsInfo,
+      str->Inputs, str->Outputs,
+      extent, 1);
+  }
+};
+
+
+
+class vtkThreadedImageAlgorithmFunctor
+{
+  vtkImageThreadStruct *str;
+  int * extents;
+  int Threads;
+  vtkExtentTranslator * ExtentTranslator;
+  bool useSmp;
+
+  
+public:
+  vtkThreadedImageAlgorithmFunctor(vtkImageThreadStruct * in,int* ext,int threads)
+  {
+    str = in;
+    extents =ext;
+    Threads = threads;
+    ExtentTranslator = vtkExtentTranslator::New();
+    
+  }
+
+  void operator()(vtkIdType begin, vtkIdType end)
+  {
+    for (int i =begin;i<=end;i++)
+    {           
+        int splitExtent[6];
+        ExtentTranslator->PieceToExtentThreadSafe(i,Threads,0,extents,&splitExtent[0],3,0);
+        str->Filter->ThreadedRequestData(str->Request,
+                                     str->InputsInfo, str->OutputsInfo,
+                                     str->Inputs, str->Outputs,
+                                     splitExtent, i);
+    }   
+  }
+};
 
 //----------------------------------------------------------------------------
 vtkThreadedImageAlgorithm::vtkThreadedImageAlgorithm()
 {
   this->Threader = vtkMultiThreader::New();
   this->NumberOfThreads = this->Threader->GetNumberOfThreads();
+  NumberOfSMPBlocks = 900;// The default number of SMP blocks.
```

Comment:
VTK code style requires that all member variables are accessed via "this->":

    this->NumberOfSMPBlocks = 900;

The reason is: the use of "this->" makes it obvious to anyone reading the code that the variable is a member variable.